jira-config/jira-config-api/src/main/java/org/xwiki/contrib/jira/config/JIRAServer.java

```diff
@@ -30,34 +32,35 @@
  */
 public class JIRAServer
 {
-    private String url;
+    private String id;
 
-    private String username;
+    private String url;
 
-    private String password;
+    private JIRAAuthenticator jiraAuthenticator;
 
     /**
      * Public-access JIRA server.
      *
      * @param url see {@link #getURL()}
+     * @param id see {@link #getId()}
      */
-    public JIRAServer(String url)
+    public JIRAServer(String url, String id)
     {
         this.url = url;
+        this.id = id;
     }
 
     /**
-     * Credential-protected JIRA server.
+     * Constructor for JIRA server which will send the requests with authentication.
      *
      * @param url see {@link #getURL()}
-     * @param username see {@link #getUsername()}
-     * @param password see {@link #getPassword()}
+     * @param id see {@link #getId()}
+     * @param jiraAuthenticator see {@link #getJiraAuthenticator()}
      */
-    public JIRAServer(String url, String username, String password)
+    public JIRAServer(String url, String id, JIRAAuthenticator jiraAuthenticator)
```

Comment:
You're breaking an API here, is it on purpose? Can't you keep the API and delegate to an authenticator?

Reply:
I agree that it's a breaking change, but from our quick search this API don't seem used a lot, so it's why we considered as it's ok to break it. But if you prefer to keep it I can do a workaround to keep it.

Reply:
Well, let's see if you can deprecate the original API while keeping it: if it's making things too complicated we'll remove it.